### PR TITLE
Abstract multiagent environment for generalizing

### PR DESCRIPTION
**Pull request by slundqui**
This PR aims to generalize our multiagent environment to allow for subclassing of a base environment for running smart fuzzing on other contracts.

## Major changes
- Adding a `base_env` class that serves as the base class for environments, and subclassing from this for our implemented environments.
- Removing random bots within environments in favor of using random agents in the runner.
- Pinning ray to 2.35
- Automatically create observation space by getting observation.
- Renaming `ray_hyperdrive_env.py` to `hyperdrive_env.py`.

### CODE REVIEW

File: traiderdaive/ray_environments/attack_hyperdrive_env.py [hunk 1]
```diff
@@ -93,29 +94,13 @@ def _get_hyperdrive_pool_config(self) -> LocalHyperdrive.Config:
...
-            Unique identifying string for the agent.
-        action: np.ndarray
-            Action activations returned by the policy network.
-
-        Returns
-        -------
-        bool
-            True if the trade was successful, False otherwise.
-        """
+    def apply_action(self, agent: LocalHyperdriveAgent, action: np.ndarray) -> bool:
```
Comment (dpaiton) [inline traiderdaive/ray_environments/attack_hyperdrive_env.py:97]: why remove the docstring?

Comment (slundqui) [reply traiderdaive/ray_environments/attack_hyperdrive_env.py:97]: Duplicate docstrings, the base class has it

File: traiderdaive/ray_environments/attack_hyperdrive_env.py [hunk 2]
```diff
@@ -50,31 +54,28 @@ class Config(RayHyperdriveEnv.Config):
...
-            {
-                agent_id: spaces.Box(
-                    low=-1e2,
-                    high=1e2,
-                    dtype=np.float64,
-                    shape=(self.env_config.num_trade_sets_per_step * self.action_length_per_trade_set,),
-                )
-                for agent_id in self.agents
-            }
+    def create_action_spce(self) -> spaces.Box:
```
Comment (wshainin) [inline traiderdaive/ray_environments/attack_hyperdrive_env.py]: Typo here - create_action_space***

**Review by wshainin — APPROVED** [review]
Looks good

**Pull request merged**